hazelcast/src/main/java/com/hazelcast/internal/partition/operation/FetchPartitionStateOperation.java

```diff
@@ -42,12 +42,11 @@ public FetchPartitionStateOperation() {
     @Override
     public void run() {
         Address caller = getCallerAddress();
-        NodeEngine nodeEngine = getNodeEngine();
-        Address master = nodeEngine.getMasterAddress();
-        if (!caller.equals(master)) {
-            String msg = caller + " requested our partition table but it's not our known master. " + "Master: " + master;
+        InternalPartitionServiceImpl service = getService();
+        if (!service.isMemberMaster(caller)) {
+            String msg = caller + " requested our partition table but it's not our known master.";
```

Comment:
should we still throw `IllegalStateException` if the caller is different than `clusterService.getMasterAddress()` here?

Reply:
yes, 👍 It seems we already have tests for both cases.